### PR TITLE
Update air-gapped installation

### LINKED ISSUE #2340: Need update document about offline-installation-tool.sh
In [Air-gapped Installation Step 4](https://kubesphere.io/docs/installing-on-linux/introduction/air-gapped-installation/#step-4-prepare-installation-images) I got information about `offline-installation-tool.sh`: 
``` sh
root@master:/home/ubuntu# ./offline-installation-tool.sh -h
Usage:
   
  ./offline-installation-tool.sh [-l IMAGES-LIST] [-d IMAGES-DIR] [-r PRIVATE-REGISTRY] [-v KUBERNETES-VERSION ]
   
Description:
  -b                     : save kubernetes' binaries.
  -d IMAGES-DIR          : the dir of files (tar.gz) which generated by `docker save`. default: ./kubesphere-images
  -l IMAGES-LIST         : text file with list of images.
  -r PRIVATE-REGISTRY    : target private registry:port.
  -s                     : save model will be applied. Pull the images in the IMAGES-LIST and save images as a tar.gz file.
  -v KUBERNETES-VERSION  : download kubernetes' binaries. default: v1.17.9
  -h                     : usage message

```
But when I download script and execute `./offline-installation-tool.sh -h`, I got: 
``` sh
Usage:

  ./offline-installation-tool.sh [-l IMAGES-LIST] [-d IMAGES-DIR] [-r PRIVATE-REGISTRY] [-v KUBERNETES-VERSION ]

Description:
  -b                     : save kubernetes' binaries.
  -d IMAGES-DIR          : the dir of files (tar.gz) which generated by `docker save`. default: /Users/rikka/Desktop/kubesphere/kubesphere-images
  -l IMAGES-LIST         : text file with list of images.
  -r PRIVATE-REGISTRY    : target private registry:port.
  -s                     : save model will be applied. Pull the images in the IMAGES-LIST and save images as a tar.gz file.
  -h                     : usage message

Examples:

# Download the default kubernetes version dependency binaries.(default: [kubernetes: v1.21.5], [helm: v3.6.3], [cni: v0.9.1], [etcd: v3.4.13])
./offline-installtion-tool.sh -b

# Custom download the kubernetes version dependecy binaries.
export KUBERNETES_VERSION=v1.22.1;export HELM_VERSION=v3.6.3;
./offline-installtion-tool.sh -b
```
There is no `-v` flag in usage, and executing the following command `./offline-installation-tool.sh -b -v v1.21.5` will repeat print usage information in console. So I think that's a mistake in this chapter.

### PR DESCRIPTION
**Pull request by Bettygogo2021**
Signed-off-by: Bettygogo2021 <91529409+Bettygogo2021@users.noreply.github.com>
Hi, folks, pls kindly help me review this doc. 3ks
/cc @kubesphere/sig-installation 
/assign @24sama @Patrick-LuoYu 
Fix #2340

### CODE REVIEW

**Comment by 24sama** [comment]
Hi @Bettygogo2021 
Thanks for these docs. I left some comments, PTAL.

**Comment by Bettygogo2021** [comment]
> Hi @Bettygogo2021 Thanks for these docs. I left some comments, PTAL.

Thx, i've updated the doc as suggested.

File: content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 1]
```diff
@@ -7,657 +7,554 @@ linkTitle: "离线安装"
...
+     - amd64
+     operatingSystems:
+     - arch: amd64
+       type: linux
+       id: centos
+       version: "7"
+       repository:
+         iso:
+           localPath: ""
+           url: #https://github.com/kubesphere/kubekey/releases/download/v2.0.0/centos-7-amd64-rpms.iso
```
Comment (24sama) [inline content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md:90]: url: "https://github.com/kubesphere/kubekey/releases/download/v2.0.0/centos-7-amd64-rpms.iso"

Comment (FeynmanZhou) [reply content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md:90]: If users are not able to download the iso file from the GitHub release (especially in CN), it there any other options?

**Comment by FeynmanZhou** [comment]
I notice there is a related [issue](https://github.com/kubesphere/website/issues/2340), we might need to verify if it has been fixed in this PR.

**Comment by Bettygogo2021** [comment]
#2340

File: content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md (hunk 1 above)
Comment (pixiake) [reply content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md]: > If users are not able to download the iso file from the GitHub release (especially in CN), are there any other options?

Users can make their own system dependency packages, which can be specified by `localPath`.

File: content/en/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 2]
```diff
@@ -1,670 +1,555 @@
...
+     ...
+     registry:
+       type: harbor
+       auths:
+         "dockerhub.kubekey.local":
+           username: admin
+           password: Harbor12345
+       plainHTTP: false
+       privateRegistry: "dockerhub.kubekey.local"
+       namespaceOverride: "kubesphereio"
```
Comment (24sama) [inline content/en/docs/installing-on-linux/introduction/air-gapped-installation.md:498]: namespaceOverride: "kubesphereio" -> namespaceOverride: ""

File: content/en/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 3]
```diff
@@ -1,670 +1,555 @@
...
+      # distributed under the License is distributed on an "AS IS" BASIS,
+      # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+      # See the License for the specific language governing permissions and
+      # limitations under the License.
+      
+      url="https://dockerhub.kubekey.local"  #Change the value of url to https://dockerhub.kubekey.local.
+      user="admin"
+      passwd="Harbor12345"
+      
+      harbor_projects=(library
```
Comment (24sama) [inline content/en/docs/installing-on-linux/introduction/air-gapped-installation.md:439]: Please refer to: https://github.com/kubesphere/ks-installer/blob/master/scripts/create_project_harbor.sh

File: content/en/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 4]
```diff
@@ -1,670 +1,555 @@
...
+         version: v1.22.0
+       ##
+       # docker-registry:
+       #   version: "2"
+       harbor:
+         version: v2.4.1
+       docker-compose:
+         version: v2.2.2
+     images:
+     - registry.cn-beijing.aliyuncs.com/kubesphereio/kube-apiserver:v1.21.5
```
Comment (24sama) [inline content/en/docs/installing-on-linux/introduction/air-gapped-installation.md:112]: Please modify this images list, I think we should use the dockerhub image.

**Comment by Bettygogo2021** [comment]
/assign @24sama 
Hi, I've updated the doc as suggested. If everything goes well, pls label it LGTM. 3ks

File: content/en/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 5]
```diff
@@ -1,670 +1,572 @@
...
+     - docker.io/kubesphere/notification-manager-operator:v1.4.0
+     - docker.io/kubesphere/notification-manager:v1.4.0
+     - docker.io/kubesphere/notification-tenant-sidecar:v3.2.0
+     - docker.io/kubesphere/elasticsearch-curator:v5.7.6
+     - docker.io/kubesphere/elasticsearch-oss:6.7.0-1
+     - docker.io/kubesphere/fluentbit-operator:v0.11.0
+     - docker.io/library/docker:19.03
+     - docker.io/kubesphere/fluent-bit:v1.8.3
+     - docker.io/kubesphere/log-sidecar-injector:1.1
+      - docker.io/elastic/filebeat:6.7.0
```
Comment (24sama) [inline content/en/docs/installing-on-linux/introduction/air-gapped-installation.md:206]: Remove an extra " " here.

File: content/en/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 6]
```diff
@@ -1,670 +1,572 @@
...
+
+  
+    {{</ notice >}}
+
+   Method 1: Run the following commands to create a Harbor project.
+
+   a. Run the following command to download the specified script to initialize the Harbor registry:
+
+      ```bash
+      curl https://github.com/kubesphere/ks-installer/blob/master/scripts/create_project_harbor.sh
```
Comment (24sama) [inline content/en/docs/installing-on-linux/introduction/air-gapped-installation.md:406]: curl -O https://github.com/kubesphere/ks-installer/blob/master/scripts/create_project_harbor.sh

File: content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 7]
```diff
@@ -7,657 +7,571 @@ linkTitle: "离线安装"
...
+   Harbor 管理员账号：**admin**，密码：**Harbor12345**。Harbor 安装文件在 **/opt/harbor** , 如需运维 Harbor，可至该目录下。
+
+    {{</ notice >}}
+
+   方法 1：执行脚本创建 Harbor 项目。
+
+   a. 执行以下命令下载指定脚本初始化 Harbor 仓库：
+
+      ```bash
+      curl https://github.com/kubesphere/ks-installer/blob/master/scripts/create_project_harbor.sh
```
Comment (24sama) [inline content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md:411]: curl -O https://github.com/kubesphere/ks-installer/blob/master/scripts/create_project_harbor.sh

File: content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 8]
```diff
@@ -7,657 +7,574 @@ linkTitle: "离线安装"
...
+   Harbor 管理员账号：**admin**，密码：**Harbor12345**。Harbor 安装文件在 **/opt/harbor** , 如需运维 Harbor，可至该目录下。
+
+    {{</ notice >}}
+
+   方法 1：执行脚本创建 Harbor 项目。
+
+   a. 执行以下命令下载指定脚本初始化 Harbor 仓库：
+
+      ```bash
+      curl -O https://github.com/kubesphere/ks-installer/blob/master/scripts/create_project_harbor.sh
```
Comment (24sama) [inline content/zh/docs/installing-on-linux/introduction/air-gapped-installation.md:414]: curl -O https://raw.githubusercontent.com/kubesphere/ks-installer/master/scripts/create_project_harbor.sh

File: content/en/docs/installing-on-linux/introduction/air-gapped-installation.md [hunk 9]
```diff
@@ -1,670 +1,572 @@
...
+
+  
+    {{</ notice >}}
+
+   Method 1: Run the following commands to create a Harbor project.
+
+   a. Run the following command to download the specified script to initialize the Harbor registry:
+
+      ```bash
+      curl -O https://github.com/kubesphere/ks-installer/blob/master/scripts/create_project_harbor.sh
```
Comment (24sama) [inline content/en/docs/installing-on-linux/introduction/air-gapped-installation.md:406]: curl -O https://raw.githubusercontent.com/kubesphere/ks-installer/master/scripts/create_project_harbor.sh

**Comment by haoke016** [comment]
While Export the artifact
```
error: Pipeline[ArtifactExportPipeline] execute failed: Module[ArtifactImagesModule] exec failed:
failed: [LocalHost] [ArtifactPullImages] exec failed after 1 retires: unknown service containerd.services.leases.v1.Leases: not implemented
```
My cluster does not use containerd

**Comment by 24sama** [comment]
> 

For now, kk export an `artifact` needs a `containerd` environment.

**Comment by 24sama** [comment]
> Can we made it as optional

For now, it's required. After kk releases v2.1.0, we don't have to rely on it.

**Pull request merged**